src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs

```diff
@@ -1211,6 +1214,88 @@ public static XmlDocument CreateJobXmlDocument(string svrName, string jobName)
             return new Tuple<SqlConnectionInfo, DataTable, ServerConnection>(sqlConnInfo, dt, serverConnection);
         }
 
+        internal async Task HandleAgentNotebooksRequest(AgentNotebooksParams parameters, RequestContext<AgentNotebooksResult> requestContext)
+        {
+            await Task.Run(async () =>
+            {
+                try
+                {
+                    var result = new AgentNotebooksResult();
+                    ConnectionInfo connInfo;
+                    ConnectionServiceInstance.TryFindConnection(
+                        parameters.OwnerUri,
+                        out connInfo);
+
+                    if (connInfo != null)
+                    {
+                        using (SqlConnection connection = new SqlConnection(ConnectionService.BuildConnectionString(connInfo.ConnectionDetails)))
+                        {
+                            connection.Open();
+                            /*
+                            This query fetches all the notebook Jobs across all the databases accessible by the user.
+                             */
+                            string getJobIdsFromDatabaseQueryString = 
+                                "DECLARE @script as varchar(MAX)\n" +
+                                "SET @script =\n" + 
+                                "'USE [?];\n" +
+                                "IF EXISTS\n" + 
+                                "(SELECT * FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = N''nb_template'')\n"+
+                                "BEGIN\n" + 
+                                "select dbo.nb_template.*,\n" +
+                                "DB_NAME() as db_name,\n" +
+                                "msdb.dbo.sysjobs.*\n" +
```

Comment:
>"msdb.dbo.sysjobs.* [](start = 32, length = 19)

Do you really need all the columns? I'd try to limit it to just the ones you need. Also keep in mind that these values could potentially be null if the job doesn't exist (say if it was deleted after running the job)

Reply:
This is still fetching all columns - I still would prefer to see these specifically listed out

---
In reply to: [302604528](https://github.com/microsoft/sqltoolsservice/pull/834#discussion_r302604528) [](ancestors = 302604528)